aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,402 @@
+        )
+        return dialogue_reference
+
+    def _extract_message_fields(
+        self,
+        message: DialogueMessage,
+        messages: Tuple[DialogueMessage, ...],
+        last_is_incoming: bool,
+    ):
+        performative = message[0]
+        contents = message[1]
+        message_id = messages.index(message) + 1
+        is_incoming, target = self._provide_unspecified_fields(
+            message, last_is_incoming=last_is_incoming, message_id=message_id,
+        )
+        return performative, contents, message_id, is_incoming, target
+
+    def prepare_skill_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[DialogueMessage, ...],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of DialogueMessages.
+        For every DialogueMessage (performative, contents, is_incoming, target):
+            - if 'is_incoming' is not provided: for the first message it is assumed False (outgoing),
+            for any other message, it is the opposite of the one preceding it.
+            - if 'target' is not provided: for the first message it is assumed 0,
+            for any other message, it is the index of the message before it in the tuple of messages + 1.
+
+        :param dialogues: a dialogues class
+        :param counterparty: the message_id
+        :param messages: the dialogue_reference
+
+        :return: the created incoming message
+        """
+        if not len(messages) >= 1:
+            raise AEAEnforceError("the list of messages must be positive.")
+
+        (
+            performative,
+            contents,
+            message_id,
+            is_incoming,
+            target,
+        ) = self._extract_message_fields(messages[0], messages, True)
+
+        if is_incoming:  # messages from the opponent
+            dialogue_reference = dialogues.new_self_initiated_dialogue_reference()
+            message = self.build_incoming_message(
+                message_type=dialogues._message_class,  # pylint: disable=protected-access
+                dialogue_reference=dialogue_reference,
+                message_id=message_id,
+                target=target,
+                performative=performative,
+                to=self.skill.skill_context.agent_address,
+                sender=counterparty,
+                **contents,
+            )
+            dialogue = cast(Dialogue, dialogues.update(message))
+            if dialogue is None:
+                raise AEAEnforceError(
+                    "Cannot update the dialogue with message number {}".format(
+                        message_id
+                    )
+                )
+        else:  # messages from self
+            _, dialogue = dialogues.create(
+                counterparty=counterparty, performative=performative, **contents
+            )
+
+        for incomplete_message in messages[1:]:  # type: ignore
```

Comment:
```suggestion
        for idx, dialogue_message in enumerate(messages[1:]):  # type: ignore
```
why is there a type ignore?